rio_tiler/models.py

```diff
@@ -187,6 +190,36 @@ def transform(self):
             else Affine.scale(self.width, -self.height)
         )
 
+    def _rescale(
+        self,
+        data: numpy.ndarray,
+        mask: numpy.ndarray,
+        in_range: Tuple[NumType, NumType],
+        out_range: Tuple[NumType, NumType] = (0, 255,),
+        out_dtype: Union[str, numpy.number] = "uint8",
+    ):
+        """Rescale data."""
+        nbands = data.shape[0]
+
+        rescale_arr = tuple(_chunks(in_range, 2))
```

Comment:
Are `in_range` and `out_range` typed correctly in the function signature?  The call to `_chunks` indicates to me these parameters can have more than 2 elements (one set of rescale values per band).

Reply:
so basically if the input is `in_range=(1,255,1,255,1,255)`  the `chunks` method will return `((1, 255), (1,255), (1,255))`. I guess the type `Tuple[NumType, NumType]` is wrong because in fact there can be more than 2 elements 🤔

Reply:
maybe switching to `Sequence[NumType]` would be better ?

Reply:
OR, maybe we let this to the user, meaning the `in_range` should always be of type
`Sequence[Tuple[NumType]]` (e.g: `((0, 255),)` ), we will still match the number of bands if only one tuple is provided. 

This will be a breaking change, but I'm fine with it because we move to a `minor` version

Reply:
You can do `Tuple[NumType, ...]` to indicate a variable length tuple

Reply:
👍 
I mean the `chunk` part is still weird and I feel it should go! 

IMO, input should be in form of `Sequence[Tuple[NumType, NumType]]`, Sequence should match the number of band or be 1.

Reply:
I agree with that, `Sequence[Tuple[NumType, NumType]]` is more explicit, as long as you are ok with the breaking change.

Reply:
@kylebarron @geospatial-jeff if we move to `Sequence[Tuple[NumType, NumType]]` do we need to validate the input of we let `type` checking do the trick?

Reply:
I don't think we need to validate, its clear from the type hint